### PR TITLE
Made changes in func.py

### PR DESCRIPTION
**Pull request by sroycho4**
# PR Description

## Related Issue

Closes #

## Checklist

- [ ] Did you add a function?
- [ ] Did you add the tests?
- [ ] Did you run your tests and are your tests passing?
- [ ] Did pre-commit not fail on any check?
- [ ] Did you follow the steps we provided?

### CODE REVIEW

**Review by sroycho4 — COMMENTED** [review]
Plz review

File: ivy/functional/frontends/torch/spectral_ops.py [hunk 1]
```diff
@@ -13,11 +12,25 @@ def bartlett_window(
     device=None,
     requires_grad=False
 ):
-    # this implementation is based on scipy.signal.windows.bartlett
```
Comment (vedpatwardhan) [inline ivy/functional/frontends/torch/spectral_ops.py:16]: Are there any failures that were originating from the `if` check in this function? If not, I think we shouldn't remove the check. Secondly, we aren't really adding docstrings for frontend functions because users can view those from the docstrings of the source framework anyway 😄

File: ivy/functional/frontends/torch/spectral_ops.py [hunk 2]
```diff
@@ -30,48 +43,3 @@ def bartlett_window(
         )
 
         return res[:-1] if periodic else res
-
-
-@to_ivy_arrays_and_back
```
Comment (vedpatwardhan) [inline ivy/functional/frontends/torch/spectral_ops.py:35]: Let's revert this change, I think these functions shouldn't be removed 😄

File: ivy/functional/frontends/torch/func.py [hunk 3]
```diff
@@ -7,7 +7,13 @@
...
-        outputs_to_ivy_arrays(ivy.vmap(fun, in_axes=in_dims, out_axes=out_dims))
-    )
+    # Wrap the input function `func` to handle native arrays
+    native_arrays_func = outputs_to_native_arrays(func)
+
+    # Apply Ivy's vectorized map operation (ivy.vmap) with specified input and output dimensions
+    ivy_vmap_result = ivy.vmap(native_arrays_func, in_axes=in_dims, out_axes=out_dims)
+
+    # Wrap the result of Ivy's vmap operation to handle Ivy arrays
+    ivy_arrays_result = to_ivy_arrays_and_back(outputs_to_ivy_arrays(ivy_vmap_result))
```
Comment (vedpatwardhan) [inline ivy/functional/frontends/torch/func.py:17]: actually the output of this line isn't an ivy array but rather a callback function, the `ivy_arrays_result` name might be a bit misleading 😅

**Review by vedpatwardhan — CHANGES_REQUESTED** [review]
Just requested a few changes, thanks for creating the PR @sroycho4 😄

**Review by sroycho4 — COMMENTED** [review]
Changes made

**Comment by vedpatwardhan** [comment]
Some of the changes still seem to be left on the PR, could you please take a look at the review comments again? Thanks @sroycho4 😄

**Comment by Ishticode** [comment]
Thank you very much for the PR. Closing this due to lack of activity with changes requested remaining unresolved for over 3 months. Please feel free to reopen if you would like to continue working on this. :)
